lib/puppet/network/http/api/master/v3/environments.rb

```diff
@@ -24,12 +24,16 @@ def call(request, response)
   private
 
   def timeout(env)
-    ttl = @env_loader.get_conf(env.name).environment_timeout
+    ttl = if Puppet.settings.set_by_config?(:environment_ttl)
+            Puppet[:environment_ttl]
+          else
+            @env_loader.get_conf(env.name).environment_timeout
+          end
+
```

Comment:
I could DRY this up with the `Puppet::Environments::Cache` to determine which timeout is active. But the `environment_timeout` setting is going away, so I didn't.